src/Rules/SyntaxErrorInQueryMethodRule.php

```diff
@@ -79,7 +79,7 @@ public function processNode(Node $node, Scope $scope): array
             return [];
         }
 
-        if ($scope->getType($args[$queryArgPosition]->value) instanceof MixedType) {
+        if ($scope->getType($args[$queryArgPosition]->value)->isSuperTypeOf(new StringType())->yes()) {
```

Comment:
I figure this one is hard to read - especially one does not immediately see that Mixed is contained. 

wdyt about adding a new method on `QueryReflection` which can be asked about a given type is resolvable/analyzable? this would also have the benefit that custom implemented rules can re-use the logic without re-implementing stuff over and over.